Update: I can get several tests to reliably fail in gotosocial's upstream test suite by running them repeatedly. Like I mentioned before, this seems like a broader issue with how the test suite runs via an in-memory sqlite database, which doesn't have the same concurrency semantics as when gotosocial runs on a real instance with sqlite, and WAL mode enabled.

With gotosocial checked out, you can pretty reliably get the test suite to fail by rerunning a test many times:

```
$ go test ./... -count=100 -run=TestStatusCreateTestSuite/TestProcessReplyToUnthreadedRemoteStatus
```

Leads to many test failures such as:

```
--- FAIL: TestStatusCreateTestSuite (0.90s)
    --- FAIL: TestStatusCreateTestSuite/TestProcessReplyToUnthreadedRemoteStatus (0.90s)
        create_test.go:276:
                Error Trace:    /home/blake/src/gotosocial/internal/processing/status/create_test.go:276
                Error:          Received unexpected error:
                                GetAPIStatus: error converting status: statusToFrontend: error converting status author: AccountToAPIAccountPublic: error counting statuses: database table is locked (262)
                Test:           TestStatusCreateTestSuite/TestProcessReplyToUnthreadedRemoteStatus
        create_test.go:277:
                Error Trace:    /home/blake/src/gotosocial/internal/processing/status/create_test.go:277
                Error:          Expected value not to be nil.
                Test:           TestStatusCreateTestSuite/TestProcessReplyToUnthreadedRemoteStatus
        panic.go:261: test panicked: runtime error: invalid memory address or nil pointer dereference
            goroutine 5830 [running]:
            runtime/debug.Stack()
                /nix/store/vxxlsbznhhkw1sb7qi5wki03d2sihpha-go-1.21.8/share/go/src/runtime/debug/stack.go:24 +0x5e
            github.com/stretchr/testify/suite.failOnPanic(0xc0a0a7cd00, {0x21370a0, 0x3ba7c80})
                /home/blake/src/gotosocial/vendor/github.com/stretchr/testify/suite/suite.go:89 +0x37
            github.com/stretchr/testify/suite.Run.func1.1()
                /home/blake/src/gotosocial/vendor/github.com/stretchr/testify/suite/suite.go:188 +0x25d
            panic({0x21370a0?, 0x3ba7c80?})
                /nix/store/vxxlsbznhhkw1sb7qi5wki03d2sihpha-go-1.21.8/share/go/src/runtime/panic.go:914 +0x21f
            github.com/superseriousbusiness/gotosocial/internal/processing/status_test.(*StatusCreateTestSuite).TestProcessReplyToUnthreadedRemoteStatus(0xc0a5080000)
                /home/blake/src/gotosocial/internal/processing/status/create_test.go:282 +0x22a
            reflect.Value.call({0xc000948880?, 0xc09d8f4a20?, 0xc0a72e1060?}, {0x23f5127, 0x4}, {0xc08f57be70, 0x1, 0xc08f57bd78?})
                /nix/store/vxxlsbznhhkw1sb7qi5wki03d2sihpha-go-1.21.8/share/go/src/reflect/value.go:596 +0xce7
            reflect.Value.Call({0xc000948880?, 0xc09d8f4a20?, 0xc0a5080000?}, {0xc08f57be70?, 0x535fad?, 0x357d5b0?})
                /nix/store/vxxlsbznhhkw1sb7qi5wki03d2sihpha-go-1.21.8/share/go/src/reflect/value.go:380 +0xb9
            github.com/stretchr/testify/suite.Run.func1(0xc0a0a7cd00)
                /home/blake/src/gotosocial/vendor/github.com/stretchr/testify/suite/suite.go:202 +0x467
            testing.tRunner(0xc0a0a7cd00, 0xc09d8cefc0)
                /nix/store/vxxlsbznhhkw1sb7qi5wki03d2sihpha-go-1.21.8/share/go/src/testing/testing.go:1595 +0xff
            created by testing.(*T).Run in goroutine 5829
                /nix/store/vxxlsbznhhkw1sb7qi5wki03d2sihpha-go-1.21.8/share/go/src/testing/testing.go:1648 +0x3ad
```

Will be digging into this more soon on the upstream side.

--- a/pkgs/servers/gotosocial/default.nix
+++ b/pkgs/servers/gotosocial/default.nix
@@ -41,7 +41,9 @@ buildGoModule rec {
   '';
 
   # tests are working only on x86_64-linux
-  doCheck = stdenv.isLinux && stdenv.isx86_64;
+  # doCheck = stdenv.isLinux && stdenv.isx86_64;
+  # checks are currently very unstable in our setup, so we should test manually for now
+  doCheck = false;
 
   checkFlags =
     let

--- a/pkgs/servers/gotosocial/default.nix
+++ b/pkgs/servers/gotosocial/default.nix
@@ -49,9 +49,6 @@ buildGoModule rec {
       skippedTests = [
         # See: https://github.com/superseriousbusiness/gotosocial/issues/2651
         "TestPage/minID,_maxID_and_limit_set"
-        # See: https://github.com/superseriousbusiness/gotosocial/pull/2760. Stop skipping
-        # this test when fixed for go 1.21.8 and above
-        "TestValidationTestSuite/TestValidateEmail"
       ];
     in
     [ "-skip=^${builtins.concatStringsSep "$|^" skippedTests}$" ];

--- a/pkgs/servers/gotosocial/default.nix
+++ b/pkgs/servers/gotosocial/default.nix
@@ -9,11 +9,11 @@ let
   owner = "superseriousbusiness";
   repo = "gotosocial";
 
-  version = "0.14.2";
+  version = "0.15.0";
 
   web-assets = fetchurl {
     url = "https://github.com/${owner}/${repo}/releases/download/v${version}/${repo}_${version}_web-assets.tar.gz";
-    hash = "sha256-3aSOP8BTHdlODQnZr6DOZuybLl+02SWgP9YZ21guAPU=";
+    hash = "sha256-vrSdFIdBcfj6+sxtvv1s/Mu85I1mKxjyUYS902oLKk4=";
   };
 in
 buildGoModule rec {
@@ -23,7 +23,7 @@ buildGoModule rec {
   src = fetchFromGitHub {
     inherit owner repo;
     rev = "refs/tags/v${version}";
-    hash = "sha256-oeOxP9FkGsOH66Uk946H0b/zggz536YvRRuo1cINxSM=";
+    hash = "sha256-z0iETddkw4C2R6ig9ZO8MTvhuWnmQ37/6q3oZ4WAzd4=";
   };
 
   vendorHash = null;

